packages/graph/hash_graph/tests/integration/postgres/mod.rs

```diff
@@ -59,374 +47,216 @@ impl DatabaseTestWrapper {
-        self.pool
-            .acquire_owned()
-            .await
-            .expect("could not acquire a database connection")
-    }
-
-    pub fn seed<D, P, L, E>(
+    pub async fn seed<D, P, L, E>(
         &mut self,
         data_types: D,
         property_types: P,
         link_types: L,
         entity_types: E,
-    ) -> Result<(), InsertionError>
+    ) -> Result<DatabaseApi<'_>, InsertionError>
     where
         D: IntoIterator<Item = &'static str>,
         P: IntoIterator<Item = &'static str>,
         L: IntoIterator<Item = &'static str>,
         E: IntoIterator<Item = &'static str>,
     {
+        let mut store = PostgresStore::new(
+            self.connection
+                .as_mut_client()
+                .transaction()
+                .await
+                .expect("Could start test transaction"),
+        );
+
+        let account_id = AccountId::new(Uuid::new_v4());
+        store
+            .insert_account_id(account_id)
+            .await
+            .expect("could not insert account id");
+
         for data_type in data_types {
-            self.create_data_type(
-                serde_json::from_str(data_type).expect("could not parse data type"),
-            )?;
+            store
+                .create_data_type(
+                    serde_json::from_str(data_type).expect("could not parse data type"),
+                    account_id,
+                )
+                .await?;
         }
 
         for property_type in property_types {
-            self.create_property_type(
-                serde_json::from_str(property_type).expect("could not parse data type"),
-            )?;
+            store
+                .create_property_type(
+                    serde_json::from_str(property_type).expect("could not parse data type"),
+                    account_id,
+                )
+                .await?;
         }
 
         // Insert link types before entity types so entity types can refer to them
         for link_type in link_types {
-            self.create_link_type(
-                serde_json::from_str(link_type).expect("could not parse link type"),
-            )?;
+            store
+                .create_link_type(
+                    serde_json::from_str(link_type).expect("could not parse link type"),
+                    account_id,
+                )
+                .await?;
         }
 
         for entity_type in entity_types {
-            self.create_entity_type(
-                serde_json::from_str(entity_type).expect("could not parse entity type"),
-            )?;
+            store
+                .create_entity_type(
+                    serde_json::from_str(entity_type).expect("could not parse entity type"),
+                    account_id,
+                )
+                .await?;
         }
-        Ok(())
-    }
 
-    pub fn create_data_type(
+        Ok(DatabaseApi { store, account_id })
+    }
+}
+impl DatabaseApi<'_> {
```

Comment:
Just for my own understanding, the reason for the struct that this is an impl for is for us to have the `accountId` injected into calls to the store?

Reply:
The struct is required as a transaction requires a reference to an actual connection. I avoided having a self-referential struct, so `DatabaseApi` is bound by the lifetime of the `DatabaseTestWrapper` lifetime. Storing the account id is a nice side-effect of this 🙂

Reply:
Gotcha, thanks!